qiskit/primitives/base/base_primitive.py

```diff
@@ -99,6 +101,17 @@ def _validate_parameter_values(
         ):
             parameter_values = (parameter_values,)
 
+        # Support Mapping
+        if isinstance(parameter_values, Mapping):
+            parameter_values = [parameter_values[parameter] for parameter in circuits[0].parameters]
+        elif isinstance(parameter_values, Sequence):
+            parameter_values = [
+                [vector[parameter] for parameter in circuit.parameters]
+                if isinstance(vector, Mapping)
+                else vector
+                for vector, circuit in zip(parameter_values, circuits)
+            ]
```

Comment:
I believe the generated `parameter_values` coming out of the first `if` should be `Sequence[Sequence]`.

Generally speaking, I would also move `# Support mapping` before `# Allow single value`, to be safe and make sure that things like this are standardize, as well as for consistency with how it is done for `np.ndarray`.

```suggestion
        # Support Mapping
        if isinstance(parameter_values, Mapping):
            parameter_values = [[parameter_values[parameter] for parameter in circuits[0].parameters]]
        elif isinstance(parameter_values, Sequence):
            parameter_values = [
                [vector[parameter] for parameter in circuit.parameters]
                if isinstance(vector, Mapping)
                else vector
                for vector, circuit in zip(parameter_values, circuits)
            ]
```

Reply:
Thank you. I forget where, but the order was sensitive, so my design was a little bad. I'll fix it later.